executor/transaction_processor.go

```diff
@@ -368,15 +333,32 @@ func (s *aidaProcessor) processRegularTx(db state.VmStateDB, block int, tx int,
 	db.SetTxContext(txHash, tx)
 	snapshot := db.Snapshot()
 	blockCtx := prepareBlockCtx(inputEnv, &hashError)
-	msgResult, err := s.applyMessage(db, msg, blockCtx, inputEnv, chainCfg, s.vmCfg)
+
+	txCtx := core.NewEVMTxContext(msg)
+	evm := vm.NewEVM(*blockCtx, txCtx, db, chainCfg, s.vmCfg)
+
+	var gasPool = new(core.GasPool)
+	gasPool.AddGas(inputEnv.GetGasLimit())
+	executionResult, err := core.ApplyMessage(evm, msg, gasPool)
+	if err != nil {
+		return res, fmt.Errorf("failed to execute transaction: %w", err)
+	}
+
+	msgResult := messageResult{
+		failed:     executionResult.Failed(),
+		returnData: executionResult.Return(),
+		gasUsed:    executionResult.UsedGas,
+		err:        executionResult.Err,
+	}
```

Comment:
I dont think we should return this error but rather use the condition on line 354 - this is key especially for ethereum-tests

This should be the correct implementation:

```suggestion
	var msgResult messageResult
	var gasPool = new(core.GasPool)
	gasPool.AddGas(inputEnv.GetGasLimit())
	executionResult, err := core.ApplyMessage(evm, msg, gasPool)
	if err != nil {
		// if transaction fails, revert to the first snapshot.
		db.RevertToSnapshot(snapshot)
		finalError = errors.Join(fmt.Errorf("block: %v transaction: %v", block, tx), err)
	} else {
		msgResult = messageResult{
			failed:     executionResult.Failed(),
			returnData: executionResult.Return(),
			gasUsed:    executionResult.UsedGas,
			err:        executionResult.Err,
		}
	}

```